### PR TITLE
[CARBONDATA-3476] Fix Read time and scan time stats in executor log for filter query

### PR DESCRIPTION
**Pull request by vikramahuja1001**
Problem: Read time and scan time stats shown wrong in executor log for filter query

Root cause: Projection read time is added in scan time because of this scan time and read time is not correct in stats

Solution: Added projection read time for both measure and dimension column in read stats

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NA
 
 - [ ] Any backward compatibility impacted? NA
 
 - [ ] Document update required? NA

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by vikramahuja1001** [comment]
@brijoobopanna @kunal642 , please review.

File: core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java [hunk 1]
```diff
@@ -246,20 +242,26 @@ private BlockletScannedResult executeFilter(RawBlockletColumnChunks rawBlockletC
...
           rawBlockletColumnChunks.getDimensionRawColumnChunks()[chunkIndex];
     }
+    //dimensionReadTime is the time required to read the data from dimension array
+    long totalReadTime = System.currentTimeMillis();
     int[][] allSelectedDimensionColumnIndexRange =
         blockExecutionInfo.getAllSelectedDimensionColumnIndexRange();
     DimensionRawColumnChunk[] projectionListDimensionChunk = rawBlockletColumnChunks.getDataBlock()
         .readDimensionChunks(fileReader, allSelectedDimensionColumnIndexRange);
+    totalReadTime = System.currentTimeMillis() - totalReadTime;
+
```
Comment (vikramahuja1001) [inline core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java]: remove this extra space

File: core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java [hunk 2]
```diff
@@ -283,25 +286,32 @@ private BlockletScannedResult executeFilter(RawBlockletColumnChunks rawBlockletC
       }
     }
 
+    long measureReadTime = System.currentTimeMillis();
     int[][] allSelectedMeasureColumnIndexRange =
         blockExecutionInfo.getAllSelectedMeasureIndexRange();
     MeasureRawColumnChunk[] projectionListMeasureChunk = rawBlockletColumnChunks.getDataBlock()
         .readMeasureChunks(fileReader, allSelectedMeasureColumnIndexRange);
+    measureReadTime = System.currentTimeMillis() - measureReadTime;
+
```
Comment (vikramahuja1001) [inline core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java]: remove this extra space

File: core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java [hunk 3]
```diff
@@ -246,20 +242,26 @@ private BlockletScannedResult executeFilter(RawBlockletColumnChunks rawBlockletC
       dimensionRawColumnChunks[chunkIndex] =
           rawBlockletColumnChunks.getDimensionRawColumnChunks()[chunkIndex];
     }
+    //dimensionReadTime is the time required to read the data from dimension array
```
Comment (vikramahuja1001) [inline core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java]: update the comment , time taken to read the projection columns after applying filtering

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/120/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3864/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/119/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/114/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/4069/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/12137/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/158/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/152/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/156/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/164/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/158/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/162/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/172/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/166/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/170/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/183/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/182/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/177/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/189/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/183/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/188/

**Comment by brijoobopanna** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/220/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/215/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/218/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**